mmf/trainers/base_trainer.py

```diff
@@ -278,16 +281,38 @@ def _forward_pass(self, batch):
         return report
 
     def _backward(self, loss):
-        self.optimizer.zero_grad()
+        # for update_frequency > 1, scale the loss so that accumulated gradients are the expected magnitude
+        loss = loss / self.config.training.update_frequency
         loss.backward()
+        self.profile("Backward time")
+
+    def _check_start_update(self, epoch_batch_index, epoch_num_batches):
+        # When epoch_num_batches is not a multiple of the effective batch size, the last few batches
```

Comment:
This is a weird case worth discussing. I decided to let the extra forward/backward passes run (see the train loop) and then we just ignore the results, which is wasteful computation. This behavior is identical to Pytorch Lightning's implementation of this feature (I think) [https://github.com/PyTorchLightning/pytorch-lightning/blob/82a20296e308a67c8d9202e4cbdf92a44b90b077/pytorch_lightning/trainer/training_loop.py].